beets/importer.py

```diff
@@ -675,10 +688,16 @@ def find_duplicates(self, lib):
 
         duplicates = []
         task_paths = {i.path for i in self.items if i}
-        duplicate_query = dbcore.AndQuery((
-            dbcore.MatchQuery('albumartist', artist),
-            dbcore.MatchQuery('album', album),
-        ))
+        keys = config['import']['duplicate_keys'].as_str().split()
+        info = self.chosen_info().copy()
+        info['albumartist'] = artist
+        album = library.Album(None, **info)
```

Comment:
It's also not entirely clear to me why we need to construct this `Album` object. Is there something that doesn't work about looking directly in `info`?

Reply:
It is to allow using flexible attributes, I added a comment